### ISSUE TITLE
Update revision detection

### ISSUE THREAD

**Issue by gunchleoc** [widelands/widelands#3469]
Update revision detection. We should still support bzr though so that we can have a Launchpad mirror for the daily PPA. From the forum discussion:

We may use the shortened hashes:

```
git rev-parse --short HEAD
```

The shortest possible hash for git is 4, but on git defaults to 7. The above mentioned command would give the following output for the current master head:

```
b3f2da404a
```

which are 10 digits (maybe that is the default on my system haven't  checked)

We have to add this to the repo config:

```
abbrev = 4
```
to get the shortest possible hash by default. 

```
b3f2
```

From what I read git automatically would extend this if the hashes aren't unique anymore.

Note: this is just how they are displayed. The stored hash is still longer:

```
b3f2da404a2cb579b3941390c98236baf7841c86
```

One addition. In my opinion the versions for GitHub/Dev builds should look like this:

```
git-<hash>-[<branch>]

```
Or

```
dev--<hash>-[<branch>]
```
That would make it easier to distinguish dev versions from builds.  If we would only use hash and branch name it would make it more difficult. ( I have my Open-Games branch in mind).

How about having a substring of the branch name, using _ and - as separators? Then let development branches start with a number matching the issue number.

**Comment by SirVer** [comment]
I am currently fighting with this for the nightly builds: since git does not contain a running number for any commit, I cannot really know if the nightlies are "behind" or "ahead" without looking at the history of the git repo. I considered

-  just adding the build time into the nightly file name, so I can distinguish when they were build.
- adding in `git rev-list --count HEAD`, which gives the number of commits on the current branch, which will also be monotonically increasing.

I settled for the latter, though it has no meaning really. It is monotonically increasing for master, so I decided on it.
